src/utils/gatewayMessageFactory.ts

```diff
@@ -0,0 +1,77 @@
+import {HealthcareProfessional} from "../models/HealthcareProfessional";
+import {Patient} from "../models/Patient";
+import {User} from "../models/User";
+import {filteredContactsFromAddresses} from "./addressUtils";
+
+export type SMSMessage = {message: string};
+export type EmailMessage = {from: string, html: string, subject: string}
+
+export interface GatewayMessageFactory<T> {
+  hcp: HealthcareProfessional;
+  link: string;
+  patient: Patient;
+
+  get(recipient: User, recipientPassword: string): T;
+}
+
+export interface EmailMessageFactory extends GatewayMessageFactory<EmailMessage> {}
+
+export interface SMSMessageFactory extends GatewayMessageFactory<SMSMessage> {}
+
+export class ICureRegistrationEmail implements EmailMessageFactory {
+  hcp: HealthcareProfessional;
+  hcpEmail: string;
+  link: string;
+  solutionName: string;
+  patient: Patient;
+
+  constructor(
+    hcp: HealthcareProfessional,
```

Comment:
Take a dataOwner instead of a HCP only

Reply:
Currently there is no dataOwner model, should I create it?

Reply:
No, let's use Typescript Multiple Type Definition : In the future, we'll have a service and a DTO dedicated to the dataOwner in the base SDK, let's not spend time for it today.